modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -402,6 +417,28 @@ protected Experiment internalGetExperiment(Application.Name appName,
     }
 
     /**
+     * Updates the tags for a given Application.
+     *
+     * @param applicationName the Application for which the tags should be added
+     * @param tags            list of tags for the experiment
+     */
+    protected void updateExperimentTags(Application.Name applicationName, Set<String> tags) {
+
+        if (tags == null)
+            tags = Collections.EMPTY_SET;
+
+        // update the tag table, just rewrite the complete entry
+        List<ExperimentTagsByApplication> listAllTags = experimentAccessor.getAllTags(applicationName.toString()).all();
```

Comment:
@AndreaSuckro I can take a guess that you might have done this to update the `experiment_tag` table if it is not consistent with the tags in the `experiment` table but it should ideally not happen anytime as tags get updated in both tables when new experiment is created or an existing experiment is modified. Is there any other flow that might update tags?

Also, in what scenario would you want to remove tags from an application (from a product requirement) because rewriting makes sense in that case too? If you want to handle deletes of tags on application, then this code in its current state does not handle that too in my opinion.

Pardon my guesses! :) I was just trying to put all information upfront because of our timezone difference. :)